internal/criocli/criocli.go

```diff
@@ -128,13 +128,23 @@ func mergeConfig(config *libconfig.Config, ctx *cli.Context) error {
 		runtimes := StringSliceTrySplit(ctx, "runtimes")
 		for _, r := range runtimes {
 			fields := strings.Split(r, ":")
-			if len(fields) != 3 {
+			// Special case 4, as runtime_type was added later and we cannot
+			// break backwards compatibility for the cli and conf.
+			if len(fields) != 3 && len(fields) != 4 {
 				return fmt.Errorf("wrong format for --runtimes: %q", r)
 			}
 			config.Runtimes[fields[0]] = &libconfig.RuntimeHandler{
 				RuntimePath: fields[1],
 				RuntimeRoot: fields[2],
 			}
+
+			// Only in case the 4th field was passed to the command-line, assume
+			// it's the RuntimeType and set it accordingly in the RuntimeHandler
+			if len(fields) == 4 {
+				config.Runtimes[fields[0]] = &libconfig.RuntimeHandler{
+					RuntimeType: fields[3],
+				}
+			}
```

Comment:
can we use a switch statement for this?

Reply:
Please, let me know if this change is what you had in mind with your suggestion.